.github/workflows/image.yml

```diff
@@ -0,0 +1,33 @@
+name: Building Base Image
+
+on:
+  pull_request:
```

Comment:
Do you really want to build an image on every update of each PR? Shouldn't the image be built only after merged to main branch?